src/compas_fab/backends/ros/backend_features/move_it_plan_motion.py

```diff
@@ -36,11 +36,13 @@ class MoveItPlanMotion(PlanMotion):
     def __init__(self, ros_client):
         self.ros_client = ros_client
 
-    def plan_motion(self, goal_constraints, start_configuration=None, group=None, options=None):
+    def plan_motion(self, robot, goal_constraints, start_configuration=None, group=None, options=None):
         """Calculates a motion path.
 
         Parameters
         ----------
+        robot : :class:`compas_fab.robots.Robot`
```

Comment:
Now that `robot` is here, can't `base_link`, `joint_names` and `joint_types` all be retrieved from it?  If so, then they wouldn't need to be a part of `options`.  I think such adjustments could be made in many of the ros backend features.

Reply:
You can see [here](https://github.com/compas-dev/compas_fab/blob/c43fb9b9ade629dd2a84c8936178f80cde008ef0/src/compas_fab/backends/ros/planner_backend_moveit.py) and [here](https://github.com/compas-dev/compas_fab/blob/c43fb9b9ade629dd2a84c8936178f80cde008ef0/src/compas_fab/backends/vrep/client.py) what it was like before the big refactoring when `robot` was still an argument to these functions.

Reply:
Done! `base_link` is still honored if the user decides to pass it as option, but now the `Robot` class needs not to know about it, so, only the MoveIt features, that can actually do something with it check if it is defined, otherwise default to using `robot` to get the base link from the root of the model.